test/Rx/Functional/Operator/ZipTest.php

```diff
@@ -0,0 +1,615 @@
+<?php
+
+namespace Rx\Functional\Operator;
+
+use Rx\Functional\FunctionalTestCase;
+use Rx\Observable\NeverObservable;
+use Rx\Testing\Recorded;
+
+class ZipTest extends FunctionalTestCase
+{
+    public function testZipNArySymmetric()
+    {
+        $e0 = $this->createHotObservable([
+            onNext(150, 1),
+            onNext(210, 1),
+            onNext(250, 4),
+            onCompleted(420)
+        ]);
+
+        $e1 = $this->createHotObservable([
+            onNext(150, 1),
+            onNext(220, 2),
+            onNext(240, 5),
+            onCompleted(410)
+        ]);
+
+        $e2 = $this->createHotObservable([
+            onNext(150, 1),
+            onNext(230, 3),
+            onNext(260, 6),
+            onCompleted(400)
+        ]);
+
+        $result = $this->scheduler->startWithCreate(function () use ($e0, $e1, $e2) {
+            return $e0->zip([$e1, $e2]);
+        });
+
+        $this->assertMessages([
+            onNext(230, [1, 2, 3]),
+            onNext(260, [4, 5, 6]),
+            onCompleted(420)
+        ],
+            $result->getMessages()
+        );
+
+        $this->assertSubscriptions([
+            subscribe(200, 420)
+        ],
+            $e0->getSubscriptions());
+
+        $this->assertSubscriptions([
+            subscribe(200, 420)
+        ],
+            $e0->getSubscriptions());
+
+        $this->assertSubscriptions([
+            subscribe(200, 420)
+        ],
+            $e0->getSubscriptions());
+    }
+
+    public function testZipNArySymmetricSelector()
+    {
+        $e0 = $this->createHotObservable([
+            onNext(150, 1),
+            onNext(210, 1),
+            onNext(250, 4),
+            onCompleted(420)
+        ]);
+
+        $e1 = $this->createHotObservable([
+            onNext(150, 1),
+            onNext(220, 2),
+            onNext(240, 5),
+            onCompleted(410)
+        ]);
+
+        $e2 = $this->createHotObservable([
+            onNext(150, 1),
+            onNext(230, 3),
+            onNext(260, 6),
+            onCompleted(400)
+        ]);
+
+        $result = $this->scheduler->startWithCreate(function () use ($e0, $e1, $e2) {
+            return $e0->zip([$e1, $e2], function ($r0, $r1, $r2) {
+                return [$r0, $r1, $r2];
```

Comment:
How about flipping the numbers here? E.g. `[$r2, $r1, $r0]`. The current selector is equal to the default one. :)